modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModalAddObjectField.tsx

```diff
@@ -157,45 +183,100 @@ const ModalAddObjectField: React.FC<IProps> = ({apiURL, observer, onClose}) => {
 						value={values.name ?? toCamelCase(values.label)}
 					/>
 
-					<Select
-						error={errors.type}
-						id="objectFieldType"
-						label={Liferay.Language.get('type')}
-						onChange={async ({target: {value}}: any) => {
-							const selectedType =
-								objectFieldTypes[Number(value) - 1];
-
-							if (selectedType === 'Picklist') {
-								const result = await Liferay.Util.fetch(
-									'/o/headless-admin-list-type/v1.0/list-type-definitions',
-									{
-										headers,
-										method: 'GET',
-									}
+					{ffObjectFieldBusinessTypeConfigurationEnabled ? (
+						<CustomSelect
+							error={errors.type}
+							label={Liferay.Language.get('type')}
+							onChange={async (type: any) => {
+								const selectObjectFieldType = objectFieldBusinessTypes.filter(
+									(objectFieldType) =>
+										objectFieldType.businessType ===
+										type.businessType
```

Comment:
You can remove this filter and reuse the function to do the same filter now, passing `type.businessType` as a parameter, like:

```js
const selectObjectFieldType = getSelectObjectFieldType(type.businessType);
```